src/ScoutingMain.elm

```diff
@@ -21,89 +23,62 @@ main =
 
 
 type Pages
-    = P1
-    | P2
+    = Page1
+    | Page2
 
 
 type Msg
-    = ScouterInput String
-    | TeamInput String
-    | MatchInput String
-    | DriverStationPositionInput String
-    | Move
+    = TeamDataMsg TeamDataView.Msg
+    | AutonomiDataMsg AutonomiDataView.Msg
+    | NextPage12
 
 
 type alias Model =
-    { scouterName : String
-    , teamNum : Int
-    , matchNum : Int
-    , driverStationPosition : String
+    { teamData : TeamDataView.Model
+    , autonomiData : AutonomiDataView.Model
     , pages : Pages
     }
```

Comment:
This way, you may have uninitialized `teamData` and `pages` that are not the first page.
The way to avoid such invalid state is to encode this requirement at the type level.
For instance, you can require each previous page data at the next page's initialization:
```elm
type alias Model
    = Setup TeamDataView.Model
    | Autonomous TeamDataView.Model AutonomousDataView.Model
```
This way, you can't create an `Autonomous` without a `TeamDataView.Model`, and the types indicates that the second page (`Autonomous`) has access to the setup page's data (although it can't do anything with it - which is precisely what we want 🙂).
Keep in mind this solution isn't perfect, because you can create a `TeamDataView.Model` easily with `TeamDataView.init`. This issue has a solution as well, but just this change makes a major improvement and is sufficient for now.